client/src/pages/UserAvailability.js

```diff
@@ -0,0 +1,165 @@
+import React from "react";
+import Onboarding from "../components/Onboarding";
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography'; 
+import Checkbox from '@material-ui/core/Checkbox';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { makeStyles } from '@material-ui/core/styles';
+import DropdownSelect from '../components/DropdownSelect';
+
+const useStyles = makeStyles((theme) => ({
+    checkBoxContainer: {
+        paddingRight:"0px !important"
+    },
+    
+    checkBox: {
+        boxShadow: "0px 1px 4px 0px rgb(0 0 0 / 15%)",
+        margin:0,
+        padding:"0px 15px 10px 15px",
+    },
+
+    checkBoxText: {
+        fontSize:"12px"
+    }
+}))
+
+const timeOptions = []
+
+for (var i = 0; i <= 23; i++) {
+    timeOptions.push(
+        {
+            value:i,
+            label:`${i}:00`
+        }
+    );
+}
+
+
+function CheckBox() {
+
+    const classes = useStyles();
+
+    const [days,setDays] = React.useState([
+        {
+            value:"Sundays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        },
+
+        {
+            value:"Mondays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        },
+
+        {
+            value:"Tuesdays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        },
+
+        {
+            value:"Wednesdays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        },
+
+        {
+            value:"Thursdays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        },
+
+        {
+            value:"Fridays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        },
+
+        {
+            value:"Saturdays",
+            state:false,
+            color:"rgb(0 0 0 / 15%)"
+        }
+    ])
+
+    const toggleCheckbox = (event) => {
+        const temp = days.slice()
+        const idx = event.target.value
+        const prevState = temp[idx].state 
+        temp[idx].state = !prevState
+        temp[idx].color = temp[idx].state ? "#000000" : "rgb(0 0 0 / 15%)"
+
+        setDays(temp)
+    }
+
+    return (
+        <FormGroup row>
+            {days.map((date,index)=>(
+                <FormControlLabel
+                key={index}
```

Comment:
Using indexes as keys is not a good idea. You could run into [problems](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318). Use something else that is unique for each element 🙂